client/head-manager.js

```diff
@@ -26,6 +27,22 @@ export default class HeadManager {
     const tags = {}
     head.forEach((h) => {
       const components = tags[h.type] || []
+
+      if (h.key) {
+        if (this.unique.has(h.key)) {
+          const c = this.unique.get(h.key)
+
+          if (components.indexOf(c) === -1) {
+            components.push(c)
+            tags[h.type] = components
+
+            return
+          }
+        }
+
+        this.unique.set(h.key, h)
+      }
```

Comment:
It is meant to cache the components that use `key` so that when switching routes it will set the cached component instead of the new one.

Example:
pages/index.js:
```
<Head>
   <link rel="stylesheet" href="/abc.css" key="test" />
</Head>
```

pages/other.js:
```
<Head>
   <link rel="stylesheet" href="/def.css" key="test" />
</Head>
```

if you navigate to `/` first and then to `/other` using a `<Link>` it will keep `/abc.css` instead of `/def.css` in this case.